examples/AeroSharp.Examples/Maps/NestedCDTMapGetByRankExample.cs

```diff
@@ -0,0 +1,57 @@
+using AeroSharp.Connection;
+using AeroSharp.DataAccess;
+using AeroSharp.DataAccess.MapAccess;
+using AeroSharp.Examples.Utilities;
+using Aerospike.Client;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AeroSharp.Examples.Maps;
+
+public class NestedCdtMapGetByRankExample : IExample
+{
+    private const string SetName = nameof(NestedCdtMapGetByRankExample);
+
+    private readonly IMap<string, object> _map;
+
+    public NestedCdtMapGetByRankExample(IClientProvider clientProvider)
+    {
+        var dataContext = new DataContext(ExamplesConfiguration.AerospikeNamespace, SetName);
+        var mapConfiguration = new MapConfiguration()
+        {
+            Order = MapOrder.KEY_VALUE_ORDERED
+        };
+        _map = MapBuilder.Configure(clientProvider)
+            .WithDataContext(dataContext)
+            .WithMapConfiguration(mapConfiguration)
+            .Build<string, object>(MapExampleData.NestedRecordKey);
+    }
+
+    public async Task ExecuteAsync()
+    {
+        await _map.DeleteAsync(cancellationToken: default);
+        await MapHydrator.HydrateMap(_map, MapExampleData.m3);
+        CTX context1 = CTX.MapKey(Value.Get("key1"));
+        CTX context2 = CTX.MapKey(Value.Get("key2"));
```

Comment:
Hmm, I'm a bit concerned that we're exposing `Value` and `CTX` here to the end-user, since that's a dependency of ours. I don't think we normally expose things directly from the client.

We may want to do some sugary stuff here. @wbaldoumas would love your thoughts as well.

Reply:
Ah I see your point.  I think the example still works without Value, I'll check on that.

CTX though - I don't want to necessarily add yet another wrapper just to add a context, but open to thoughts!  It was more of a - oh look this works for now sorta thing. Didn't put much thought into it though so I appreciate the perspective!

Reply:
Yeah, I understand the reluctance to create a wrapper for the sake of doing it-- I just think it's important for us to think about it a little bit specifically.

Looking at the [CTX docs](https://docs.aerospike.com/apidocs/csharp/html/t_aerospike_client_ctx) I do see where it provides methods like `FromBase64`, `ListIndex` (related to list implementations, which maybe we should support CTX style stuff for list operations), `ListRank`, etc.

I think it _could_ be relevant to provide a reduced set of methods for Map-specific CTX stuff.

We should probably create an issue to support this for lists also.

Reply:
@farmerau I was hesitant to expose both the `CTX` and `MapReturnType` in my initial PR adding map support because I had similar sentiments as you. It feels against the spirit of this library to make users think about the lower level details like that. That said, we _could_ introduce `CTX` and make it optional (or create method overloads to keep the existing signatures small).

@awesor in order to get the functionality that you need out of this lib, is exposing the `CTX` necessary, or is there some way we could make a happier abstraction for that?

Reply:
I think, if you guys give me a bit of time here, I can definitely get some abstractions setup!

Just to clarify, for my own sake here, are we in agreement that the `params` keyword isn't necessary and at the end of the day, we just need to be able to pass in a list of CTX-typed objects to the operations?

Are we okay with `MapReturnType` sticking around?

Reply:
I feel pretty confident about `params` being unnecessary.

Is `MapReturnType` actually being exposed anywhere? As far as I know, it's isolated in the `MapOperation` class, which is internal to the library. https://github.com/wayfair-incubator/AeroSharp/blob/65f8c4ebfcb98984956694fe8bdf72bbdaaceb6d/src/AeroSharp/DataAccess/Internal/MapOperations.cs

Reply:
@farmerau that's right - we aren't exposing `MapReturnType` anywhere. My preference is that if users want different return types (just keys, just values, etc.), we implement separate methods for them to invoke so we can keep the API interfaces simple.

Reply:
I think (for now) that means that @awesor doesn't need to touch anything around that arg here.